### PR TITLE
ci: AppVeyor: ensure that win32 feature is set

### PR DESCRIPTION
**Pull request by blueyed**
[skip travis]

### CODE REVIEW

File: ci/build.ps1 [hunk 1]
```diff
@@ -107,6 +107,9 @@ cmake -G $cmakeGenerator $(convertToCmakeArgs($nvimCmakeVars)) .. ; exitIfFailed
 cmake --build . --config $cmakeBuildType -- $cmakeGeneratorArgs ; exitIfFailed
 bin\nvim --version ; exitIfFailed
 
+# Ensure that win32 feature is set.
+bin\nvim -c 'exe has("win32") ? "q" : "cq"' ; exitIfFailed
```
Comment (justinmk) [inline ci/build.ps1]: Nvim cq supports controlling the exit code.

    nvim -c "exe has('win32').'cq'" ; exitiffailed

Comment (blueyed) [reply ci/build.ps1]: Thanks, amended.

**Comment by blueyed** [comment]
Why is this hanging on CI?
https://ci.appveyor.com/project/neovim/neovim/builds/25266974/job/9ukxi8qxx5mnr5rt#L2310

**Comment by blueyed** [comment]
@janlazo 
Any idea why this hangs on CI?  Does it work on Windows by itself?
https://ci.appveyor.com/project/neovim/neovim/builds/25291260/job/m99yd0vwgxyhgogi#L6000

**Comment by janlazo** [comment]
Got exit code `-1073741701`. Perhaps update `exitIfFailed` to echo `$LASTEXITCODE` before exiting the script.

File: ci/build.ps1 [hunk 2]
```diff
@@ -107,6 +107,9 @@ cmake -G $cmakeGenerator $(convertToCmakeArgs($nvimCmakeVars)) .. ; exitIfFailed
 cmake --build . --config $cmakeBuildType -- $cmakeGeneratorArgs ; exitIfFailed
 bin\nvim --version ; exitIfFailed
 
+# Ensure that the "win32" feature is set.
+bin\nvim -u NONE --headless -c 'exe !has("win32")."cq"' ; exitIfFailed
```
Comment (janlazo) [inline ci/build.ps1]: Escape with double quotes with a backslash.

Comment (janlazo) [reply ci/build.ps1]: https://github.com/janlazo/dotvim8/commit/a2855dc59e85a3c801a78776d534581924f8337f#diff-dad8a9269e73f782818bbba0843425f0R44

Comment (blueyed) [reply ci/build.ps1]:
```
bin\nvim -u NONE --headless -c 'exe !has(\"win32\").\"cq\"' ; exitIfFailed
```
?

Would the ampersand help here?  (but I think I've tried that in some variant)
`& bin\nvim -u NONE --headless -c "exe !has('win32').\"cq\"" ; exitIfFailed`

I assume you have Windows available at your hands?

Comment (janlazo) [reply ci/build.ps1]: `.\bin\nvim.exe` is enough. powershell will autocomplete it like this even if you can run it with `./bin/nvim.exe`.
ampersand is useful for executing escaped paths.
`& "$NVIM_ABS_PATH" -u NONE --headless ...`

Comment (blueyed) [reply ci/build.ps1]: > `.\bin\nvim.exe` is enough

What do you mean?  Just adding a dot in front here should fix it?

Comment (janlazo) [reply ci/build.ps1]: Prepending with `.\` to run in the current directory. I got a warning in the default powershell terminal when I ran nvim without it.

Comment (blueyed) [reply ci/build.ps1]: Ok.  But `bin\nvim --version` above does not hang.

**Comment by blueyed** [comment]
I give in...

@b-r-o-c-k 
Can you help here maybe?

**Review by janlazo — APPROVED** [review]
I'm ok with this as a sanity check for `--headless`.

**Pull request merged**